src/app/base/components/NodeConfigurationFields/NodeConfigurationFields.tsx

```diff
@@ -50,14 +50,15 @@ const NodeConfigurationFields = (): JSX.Element => {
             name="description"
           />
           <TagIdField
+            addTagRef={ref}
```

Comment:
Could you explain what happened here, why is this necessary?

Reply:
The normal implementation calls for the event object to be passed to the `openPortal` function, in a situation where this can't be done, the alternative is to pass a ref as shown [here](https://www.npmjs.com/package/react-useportal#usage-with-a-ref). Without the ref, it throws an error in the console (attached below)
![image](https://github.com/canonical/maas-ui/assets/47540149/90de1cae-4401-43fc-b5eb-9d7f06000543)

Reply:
I can see you added the ref to the `li` element.

It's not an interactive element.`ref` needs to be added to an element you're _interacting with_ (that triggers the portal), in this case it's a add new tag _button_. This is for many reasons, one of which accessibility.

I'd recommend to modify the existing `onAddNewTag` to include the event so that it can be passed to the openPortal handler, e.g.:

```tsx
onAddNewTag={(name) => (event) => {
  setNewTagName(name);
  openPortal(event);
 }}
```
By using a curried function, you ensure that any handler that doesn't care about the event will still work as expected. 

That way, you can avoid passing down the ref entirely.